File: labs/lib/typescript/porto_client.ts
/** Lab PortoClient factory. Injects tracing transport when HTTP trace is enabled. */

import { PortoClient, type PortoConfig } from "../../../sdks/porto-sdk-typescript/src/client.ts";
import { resolveTransport } from "../../../sdks/porto-sdk-typescript/src/config.ts";
import { HttpClient } from "../../../sdks/porto-sdk-typescript/src/transport/http-client.ts";
import { wrapTransport } from "./http_trace.ts";

export function createPortoClient(config: PortoConfig): PortoClient {
  const policy = resolveTransport(config);
  const inner = new HttpClient(policy.timeout, policy.retries);
  return new PortoClient(config, { transport: wrapTransport(inner) });
}

File: labs/lib/python/porto_client.py
"""Lab PortoClient factory. Injects tracing transport when HTTP trace is enabled."""

from __future__ import annotations

from porto_sdk import PortoClient, PortoConfig
from porto_sdk.transport import HttpClient

from .http_trace import wrap_transport


def create_porto_client(config: PortoConfig | None = None) -> PortoClient:
    cfg = config or PortoConfig()
    policy = cfg.resolved_transport()
    inner = HttpClient(timeout=policy.timeout, retries=policy.retries)
    return PortoClient(cfg, transport=wrap_transport(inner))
